### DISCUSSION TITLE
Ender 3 V2 with BLtouch Z axis will not decrement

### DISCUSSION THREAD

**Discussion by boulderlauren** [mriscoc/Ender3V2S1#1273] [Q&A] [answered]
main board 4.2.2
bl touch V3.1
servo driver TMC2208
TJC LCD screen

firmware Ender3V2-422-BLTUBL-MPC-20240125.bin
BLtouch ver 3.1

The Z-axis servo seems disabled.  It only moves in the positive direction and will not go all the way down to the bed.  I feel like there is a fault tripped somewhere and I can't figure out where.  I've got all source and am looking through it to figure out why I can't get things to work.  

Suring auto home BL goes red then moves to X Y limit then center of bed then pin is pushed out twice then BL turns blue and moves in Z pos direction.  Every time I home Z the hot end is moved up in the X+ direction and I can't get it to move downwards. even using Pronterface com.

Any ideas?

**Comment by mlask** [comment]
Z-axis cannot be moved below 0 value - if printer correctly homes Z then this position is set as "0". You'll need to set Z offset to move "below" 0 (in fact, it just lowers the 0 physical position).

**Comment by boulderlauren** [comment]
I can not home the Z axis successfully as it fails which means the printer doesn't know where it is in space.  When I set the Z-offset to  a negative number there is no change in behavior.  I feel I have tripped a software fault for the z axis.  when M999 is employed on pronterface it doesn't clear the error.  

Could the BLtouch be broken?  I checked all wiring and continuity and everything is working.
I ordered a new bltouch i the meantime.
I even compiled my new firmware and tried multiple sources for firmware:  E3V2-BLTouch-5x5-v4.4.4-v2.01.bin, 
Ender3V2-422-BLTUBL-MPC-20240125.bin to see if they were the issue.

I can't find where in the Marlin code a fault is being tripped so I can create the proper conditional statement or disable the error.

No bed crashing is going on .  
Pin deploys and resets on bltouch correctly
Z axis limiter switch is unplugged

**Reply by mriscoc** [reply]
The probe is wiring incorrectly or in an incompatible way. Using the color of the original BLTouch wiring as a reference, check the white and black wires.

**Reply by boulderlauren** [reply, marked as answer]
brown red yellow black white
G V In G Out

on the board connector and BL touch. Continuity checked but now I'm soldering into the pins and will reassemble connector and retest continuity. BTW I like your firmware which I have compiled my own version as I want to play with it.

From the moment I boot the Ender I can't home or move downwards
Where in the code is the z axis error triggered? Initialization? Is that why you think it is the wiring?

I had it wired wrong but it was fixed and that is why I think the sensor is bad and why I'm using a better connector with soldered wiring just in case it is poor connections.

**Reply by boulderlauren** [reply]
Mriscoc was right and in the end it took me spending some time creating good connections, soldering the pins and now everything is working

**Comment by boulderlauren** [comment]
brown red yellow black white
G V In G + Out

on the board connector and BL touch.  Continuity checked but now I'm soldering into the pins and will reassemble connector and retest continuity.

**Reply by boulderlauren** [reply]
G V In G Out
 above is wrong

**Comment by boulderlauren** [comment]
BTW I like your firmware which I have compiled
From the moment I boot the Ender I can't home or move downwards
Where in the code is the z axis error triggered?  Initialization?  Is that why you think it is the wiring?

I had it wired wrong but it was fixed and that is why I think the sensor is bad.
